static/js/vue-cdr-access/src/mixins/routeUtils.js

```diff
@@ -0,0 +1,82 @@
+import isEmpty from 'lodash.isempty';
+
+export default {
+    methods: {
+        /**
+         * Put URL parameters into an object
+         * Set default params, if none are present that can be updated
+         * @param params_to_update
+         * @returns {({} & Dictionary<string | (string | null)[]>) | {start: number, page: number, sort: string, rows: number}}
+         */
+        urlParams(params_to_update = {}) {
+            let route_params = Object.assign({}, this.$route.query);
+            let page_params;
+
+            if (isEmpty(this.$route.query)) {
+                page_params = {
+                    page: 1,
+                    rows: 20,
+                    start: 0,
+                    sort: 'title,normal'
+                };
+            } else if (!isEmpty(params_to_update)) {
+                page_params = Object.assign(route_params, params_to_update);
+
+                if (!this.paramExists('format', params_to_update)) {
+                    delete page_params.format;
+                }
+            } else {
+                page_params = route_params;
+            }
+
+            return page_params;
+        },
+
+        /**
+         * Expects an object of parameters. Generally used with urlParams() method above
+         * @param params
+         * @returns {string}
+         */
+        formatParamsString(params) {
```

Comment:
this approach seems okay, but wanted to leave it up to you if you'd rather use any of the suggestions in this thread:
https://stackoverflow.com/questions/5505085/flatten-a-javascript-object-to-pass-as-querystring
They all seem to encode the param name as well, which might be a good idea. There are lodash and es6 approaches in here.

Seems like the vue router can do this as well, but that would only be useful if you were going through the router. Not sure if that is practical here:
`this.$router.push({ query: { kw: kw, p: 1 }})`